src/main/java/fi/aalto/cs/apluscourses/intellij/model/SettingsImporterImpl.java

```diff
@@ -91,11 +104,54 @@ public void importProjectSettings(@NotNull Project project, @NotNull Course cour
 
     for (String fileName : fileNames) {
       Path path = Paths.get(fileName);
+      // The ZIP contains a .idea directory with all of the settings files. We want to extract the
+      // files to the .idea directory without the .idea prefix, as otherwise we would end up with
+      // .idea/.idea/<settings_files>.
       Path pathWithoutRoot = path.subpath(1, path.getNameCount());
-      zipFile.extractFile(path.toString(), settingsPath.toString(), pathWithoutRoot.toString());
+      zipFile.extractFile(path.toString(), target.toString(), pathWithoutRoot.toString());
     }
+  }
+
+  private static final String WORKSPACE_XML_COMPONENT_NAME = "CompilerWorkspaceConfiguration";
+  private static final String WORKSPACE_XML_OPTION_NAME = "AUTO_SHOW_ERRORS_IN_EDITOR";
+  private static final String WORKSPACE_XML_OPTION_VALUE = "false";
 
-    ProjectManager.getInstance().reloadProject(project);
+  /**
+   * Parses the XML file (usually the workspace.xml file stored in .idea) at the given path and
+   * returns a {@link Document} with file contents and an additional setting added.
+   * @param workspaceXmlPath The path pointing to the XML file.
+   */
+  @NotNull
+  public static Document createCustomWorkspaceXml(@NotNull Path workspaceXmlPath) throws
+      IOException {
+    try {
+      Document document = DomUtil.parse(workspaceXmlPath.toFile());
+      Node projectNode = document.getDocumentElement();
+
+      // Check if a component with the given name already exists
+      List<Node> componentMatches = DomUtil.getNodesFromXPath(
+          "//component[@name=\"" + WORKSPACE_XML_COMPONENT_NAME + "\"]", projectNode);
+      Node compilerConfigurationNode;
+      if (componentMatches.isEmpty()) {
+        Element compilerConfigurationElement = document.createElement("component");
+        compilerConfigurationElement.setAttribute("name", WORKSPACE_XML_COMPONENT_NAME);
+        compilerConfigurationNode = compilerConfigurationElement;
+        projectNode.appendChild(compilerConfigurationNode);
+      } else {
+        // There should only be one match, otherwise the workspace.xml file is in a weird state
+        compilerConfigurationNode = componentMatches.get(0);
+      }
```

Comment:
can this be, for example, a separate method?

Reply:
In this case I think it would probably add more complexity. We add `compilerConfigurationNode` to `projectNode` if it doesn't exist yet, and I think it's easier to follow the logic this way